### ISSUE TITLE
pek average... 

### ISSUE THREAD

**Issue by andreazammit** [IALSA/IALSA-2015-Portland#162]
Hi @ampiccinin and @andkov can you check these two models out? I have attempted the PEK average, rather than best of three trials, but the outcome is looking funny, esp for the females. I increased iterations and it did nothing for them. Could this be because of a lot of missing data? Or maybe I am missing something ?Thanks!

[b1_female_aehplus_pulmonary_memory_pek_freerecall.pdf](https://github.com/IALSA/IALSA-2015-Portland/files/635335/b1_female_aehplus_pulmonary_memory_pek_freerecall.pdf)
[b1_male_aehplus_pulmonary_memory_pek_freerecall.pdf](https://github.com/IALSA/IALSA-2015-Portland/files/635336/b1_male_aehplus_pulmonary_memory_pek_freerecall.pdf)

**Comment by ampiccinin** [comment]
hmmm, yes, @andreazammit  - you have an error message that a line in your input file is longer than 90 characters, and a bunch of people are apparently missing time scores, but the biggest problem (possibly related to the first, but I'm not sure) is your P5 variable. Mplus is reading something other than P5 (possibly SubjectID?).  

Maybe start by just looking at P5 in your data file. Maybe an error was made when calculating it? 

If not, then try fixing c1= FreeRecall_1; c2= FreeRecall_2; c3= FreeRecall_3; c4=
FreeRecall_4; c5= FreeRecall_5    so it does not wrap at the end of the line, but has a hard return. ALSO, please add a semi-colon after FreeRecall_5;

Good luck!

**Comment by andreazammit** [comment]
@ampiccinin I think the problem was pekavg_5... The figures are fine, but I think there were too many missing values. I re-rean the models with up to 4 waves, and they are looking better. Do you think we can go ahead and run all pekavg with just 4 waves? 
[b1_male_aehplus_pulmonary_memory_pek_freerecall.pdf](https://github.com/IALSA/IALSA-2015-Portland/files/636728/b1_male_aehplus_pulmonary_memory_pek_freerecall.pdf)
[b1_female_aehplus_pulmonary_memory_pek_freerecall.pdf](https://github.com/IALSA/IALSA-2015-Portland/files/636729/b1_female_aehplus_pulmonary_memory_pek_freerecall.pdf)

**Comment by ampiccinin** [comment]
You could do that. I can't see the n at wave 5 in the output, but I would check whether the wave 5 values are odd (large) before entering Mplus and whether the data are being read into Mplus correctly first because of the really large mean. What do the ID#s look like? Are they really large numbers?

**Comment by andreazammit** [comment]
Hi Andrea, the N at wave 5 is 329 with a mean of 265.8. I don't understand why it would read the subject ID? I check the script over and over and there doesn't seem to be anything odd... The Subject IDs are large though as you predicted. Do you know why it would estimate IDs rather than the variable that I specified? 
Thanks so much!

**Comment by ampiccinin** [comment]
:) I was imagining that somehow you have one fewer variables in your dataset than in your Mplus variable list so that Mplus is reaching over and grabbing the next variable to fill the gap. Now that I think about this again, it is probably not correct, since then I think everything would be messed up, not just FR_5. Sometimes weird stuff can happen. Mplus is definitely not reading the data for FR_5 in correctly and there are a limited number of ways this can happen. Can you check your variable list in the new dataset containing PEK_avg against your Mplus variable list?

**Comment by andreazammit** [comment]
Yes, checked. I just added an additional five variables to the list at the very back. So it really should not have been a problem. To make sure it's not a problem with the last column, I deleted one variable that I was not using from the list to push things up, and likewise deleted it from the MPlus script, and I still got the same results. 
I ran the models using a different cognitive variable, logical memory, and again with 5 waves I run into a lot of problems for both males and females, but with four, it seems to run smoothly, so I do think the problem is the pekavg_5, although I can't figure out what or why it is doing this given the means and the n is not troublesome and everything is in its correct place...

**Comment by ampiccinin** [comment]
That sure is a mystery. 

Crazy idea - what if you add another variable after pekavg_5, so it is not the last one in the file? 

(up to you. If you are fed up and just want to go with 4 waves, let me know.)

**Comment by andreazammit** [comment]
I added two different variables as a last (i.e. so that PEK is not last variable) in the file and I got this output twice for the two different variables. It seems the p5 is the problem..... I think I am just going to stick to 4 waves.... Unless you have any other ideas? But P5 definitely has more than 4 observations, and I did change the script in mplus too because I double checked that the last variable is changed accordingly to the updated file...... 

*** ERROR
  One or more variables have a variance of zero.
  Check your data and format statement.

    Continuous    Number of
     Variable   Observations    Variance

      P1             130        4777.938
      P2              82        6542.484
      P3              61        5305.871
      P4              46        4343.485
    **P5               4           0.000
      C1             159          43.994
      C2             156          42.152
      C3             107          46.487
      C4              74          47.534
      C5              54          41.827
      HEIGHT         130          59.907
      BAGE           159          27.869
      EDUC           159          10.943
      DIAB           159           0.166
      SMOKHIST       159           0.704
      CARDIO         159        5342.622

*** WARNING
  Data set contains cases with missing on x-variables.
  These cases were not included in the analysis.
  Number of cases with missing on x-variables:  441

**Comment by ampiccinin** [comment]
...I was about to say "OK - let's go with 4" when I noticed that the CARDIO variance is crazy too. (or now? Was it before?)

So - let's do this: in whatever software you are using outside Mplus, save a file that contains ONLY the variables you need for the various Portland projects (or even just for this one). This will give you a smaller file to work with that, ideally, will make it easier to figure out where things are going wrong.

**Comment by andreazammit** [comment]
Hi @ampiccinin sorry to be slow with this, but I just minimized the data to suit pekaverage. Think the output is looking much better now - are you happy to go on with this format for the rest of the pek models? Thanks a lot!
[b1_female_aehplus_pulmonary_memory_pek_logicalmemory.pdf](https://github.com/IALSA/IALSA-2015-Portland/files/664080/b1_female_aehplus_pulmonary_memory_pek_logicalmemory.pdf)
[b1_male_aehplus_pulmonary_memory_pek_logicalmemory.pdf](https://github.com/IALSA/IALSA-2015-Portland/files/664081/b1_male_aehplus_pulmonary_memory_pek_logicalmemory.pdf)

**Comment by ampiccinin** [comment]
@andreazammit  - WOW!!! 

While I am anxious to complete this project, I am also patient and determined to feel confident in the results we present. I really appreciate the time you are spending on this and have been worried that we have been asking too much from you. 

This definitely looks better. Definitely different too. Part of what is surprising me the most is that your N is now 1068 (for women, 690 for men), rather than 150 (72 for men)!  Almost all the SEs have shrunk - sometimes by 50%! - except, interestingly, the SP-SC variances and correlation...

For some reason - and I can't see what is different in your Input file other than requesting the variance of height (height;) - we aren't getting the UNIVARIATE HIGHER-ORDER MOMENT DESCRIPTIVE STATISTICS, which shows us the n for each variable. The "Covariance Coverage" section tells us that height data are only available on around 42% of participants, but out of 1068, that is still WAY more than 150. The means and covariances in the ESTIMATED SAMPLE STATISTICS look fine (and would be basically redundant to what is printed in the missing output).

Here are two, I hope final, requests (one simple and super-quick, one not too time-consuming):

1) please check whether there are spaces between MMS_1 MMS_2, FAS_1 FAS_2, and CAT_1 CAT_2. My cursor says there are, but my eyes are not so sure. The output would certainly look weirder if Mplus thought MMS_1MMS_2 was a single variable, so there almost certainly is a space for each of these, but this is super easy to check, so we might as well, given the weirdness in the prior file.

2) Check the descriptives from the Mplus output against those in the software you used to create the file (SPSS?). If they agree, then let's go ahead with the other models!!!!

YAYYYY!!!!!
